basis_universal.lua

```diff
@@ -0,0 +1,631 @@
+project "basis_universal"
+
+local prj = project()
+local prjDir = prj.basedir
+
+-- -------------------------------------------------------------
+-- project
+-- -------------------------------------------------------------
+
+-- common project settings
+
+dofile (_BUILD_DIR .. "/3rdparty_static_project.lua")
+
+-- project specific settings
+
+uuid "E2587158-A6F8-4BF3-9A7D-9CB1C2F0A5FE"
+
+flags {
+  "NoPCH"
+}
+if not (_PLATFORM_WINDOWS or _PLATFORM_WINUWP) then
+  buildoptions {
+    "-Wall",
+    "-Wno-reorder",
+    "-Wno-unknown-pragmas",
+    "-Wno-undefined-inline",
+    "-Wno-unused-variable",
+  }
+end
+
+files {
+  "encoder/**.h",
+  "encoder/**.cpp",
+  "transcoder/**.h",
+  "transcoder/**.cpp",
+  "*.h",
+  "*.cpp",
+}
+
+excludes {
+  "i3s-lib/src/dllmain.cpp",
+}
+
+includedirs {
+  "encoder",
+  "transcoder",
+}
+
+defines {
+}
+
+-- @see end of script to disable warnings
+-- -------------------------------------------------------------
+-- configurations
+-- -------------------------------------------------------------
+
+if (_PLATFORM_WINDOWS) then
+  -- -------------------------------------------------------------
+  -- configuration { "windows" }
+  -- -------------------------------------------------------------
+
+  -- common configuration settings
+
+  dofile (_BUILD_DIR .. "/3rdparty_static_win.lua")
+
+  -- project specific configuration settings
+
+  configuration { "windows" }
+
+  defines {
```

Comment:
Is pplx used in this project? It might be better to start with a cleaner lua file that doesn't bring in unneeded commented out lines.